db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -0,0 +1,188 @@
+#
+# Description: <Method description here>
+#
+def request_info
+  @service = ($evm.root['vmdb_object_type'] == 'service_template_provision_task') ? true : false
+  @miq_request = $evm.root['miq_request']
+  $evm.log(:info, "Request: #{@miq_request.description} id: #{@miq_request.id} ")
+end
+
+def calculate_requested(options_hash = {})
+  {:memory  => get_total_requested(options_hash, :vm_memory),
+   :cpu     => get_total_requested(options_hash, :cores_per_socket),
+   :storage => get_total_requested(options_hash, :storage),
+   :vms     => get_total_requested(options_hash, :number_of_vms)}
+end
+
+def get_total_requested(options_hash, prov_option)
+  total_requested = collect_template_totals(prov_option)
+  total_requested = request_totals(total_requested.to_i,
+                                   collect_dialog_totals(prov_option, options_hash).to_i) if options_hash
+  total_requested
+end
+
+def request_totals(template_totals, dialog_totals)
+  template_totals < dialog_totals ? dialog_totals : template_totals
+end
+
+def collect_template_totals(prov_option)
+  total = collect_totals(service_prov_option_value(prov_option)) if @service
+  total = collect_totals(vm_prov_option_value(prov_option)) unless @service
+  total
+end
+
+def get_option_value(request, option)
+  request.get_option(option).to_i
+end
+
+def service_prov_option_value(prov_option, options_array = [])
+  @service_template.service_resources.each do |child_service_resource|
+    if @service_template.service_type == 'composite'
+      composite_service_options_value(child_service_resource, prov_option, options_array)
+    else
+      next if @service_template.prov_type.starts_with?("generic")
+      options_value(prov_option, child_service_resource.resource, options_array)
+    end
+  end
+  options_array
+end
+
+def vm_prov_option_value(prov_option, options_array = [])
+  number_of_vms = get_option_value(@miq_request, :number_of_vms)
+  case prov_option
+  when :vm_memory
+    memory_in_request = number_of_vms * get_option_value(@miq_request, :vm_memory)
+    set_requested_value(prov_option, memory_in_request,
+                        @miq_request.get_option(:instance_type), options_array)
+  when :number_of_cpus
+    requested_number_of_cpus(number_of_vms, options_array)
+  when :storage
+    requested_storage(prov_option, number_of_vms, options_array)
+  else
```

Comment:
@tinaafitz 
Do we need a log message here to indicate that it is one of the 'unknown' types.